pkgs/applications/graphics/processing/default.nix

```diff
@@ -1,73 +1,94 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, fetchurl, xmlstarlet, makeWrapper, ant, jdk, rsync, javaPackages, libXxf86vm, gsettings-desktop-schemas }:
+{ lib, stdenv, fetchFromGitHub, fetchurl, ant, unzip, makeWrapper, jdk11, javaPackages, rsync, gsettings-desktop-schemas, xorg }:
+let
+  vaqua = fetchurl {
+    name = "VAqua9.jar";
+    url = "https://violetlib.org/release/vaqua/9/VAqua9.jar";
+    sha256 = "cd0b82df8e7434c902ec873364bf3e9a3e6bef8b59cbf42433130d71bf1a779c";
+  };
+
+  jna = fetchurl {
+    name = "jna-5.8.0.zip";
+    url = "https://github.com/java-native-access/jna/archive/5.8.0.zip";
+    sha256 = "7cRkkxyo2Ld19Nif8oWceITt7dYMUnQlrs0R5e0AmPE=";
+  };
+
+  gluon = fetchurl {
+    name = "javafx-16-sdk-linux.zip";
+    url = "https://gluonhq.com/download/javafx-16-sdk-linux";
+    sha256 = "W9Q5LQ0tak3cEr7fjZEtSvF/hiLwVg2+AoH1bE+e0Uw=";
+  };
+
+  batik = fetchurl {
+    name = "batik-bin-1.14.zip";
+    url = "https://archive.apache.org/dist/xmlgraphics/batik/binaries/batik-bin-1.14.zip";
+    sha256 = "vNYa4jW7QSPucGHKFWmLI2eTOnRYfmzDNzUiwmzCX28=";
+  };
 
+  ffmpeg = fetchurl {
+    url = "https://download.processing.org/ffmpeg/4.4/ffmpeg-linux-amd64.gz";
+    name = "ffmpeg-4.4.gz";
+    sha256 = "45M5oArfpTuGONyeem9+a2bXCrfw7TEILoZ4znXrcPo=";
+  };
+in
 stdenv.mkDerivation rec {
   pname = "processing";
-  version = "3.5.4";
+  version = "1276-4.0b1";
```

Comment:
Isn't it version 4.0?

Reply:
I am not sure i understand. Yes, it is indeed version 4, as written in the string. The version comes from the revision in the processing/processsing4 repository, which is "1276-4.0b1". Should i change it to just `4.0b1`? Or `4.0b1-1276`? (I reuse the version string as the revision string)

Reply:
That you reuse the version string is perfectly fine, I was referring to your commit/PR title which say it's 4 without the commit, and release candidate numbers included. But I think it's fine, was just initially confused...

Reply:
For what it's worth, I think it makes sense to change it to `4.0b1`.

If you look at the Processing 3 repo, releases are named:
- `3.5.4`
- `3.5.3`
- `3.5.2`
- etc.

but the release information and git tag also contain the revision, e.g. `3.5.4`:
- git tag: `processing-0270-3.5.4`
- release header: `PROCESSING 3.5.4 (REV 0270) - 17 January 2020`
- [link to release](https://github.com/processing/processing/releases/tag/processing-0270-3.5.4)

This is also consistent with the Processing.org download links, the version being `processing-4.0b1.zip`.

Reply:
Also I think we should add this as `processing4` and leave `processing` alone until Processing 4 becomes stable. And keep a `processing3` package due to many breaking changes between the two.

Reply:
https://processing.org/download/ lists 4.0 beta 2 as a stable release

Reply:
Also, there is an alias `processing3` to `processing`, that may need to be updated if the `processing` package is for Processing 4.